app.js

```diff
@@ -1,8 +1,128 @@
-// import functions and grab DOM elements
+const currentPollDiv = document.getElementById('current-poll-div');
+const newPoll = document.getElementById('new-poll');
+const previousPollsDiv = document.getElementById('previous-polls-div');
+const savePollBtn = document.getElementById('save-poll-btn');
 
-// let state
+const optionOnePlus = document.getElementById('option-one-plus-btn');
+const optionOneMinus = document.getElementById('option-one-minus-btn');
+const optionTwoPlus = document.getElementById('option-two-plus-btn');
+const optionTwoMinus = document.getElementById('option-two-minus-btn');
 
-// set event listeners 
-  // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+let question = '';
+let optionOne = '';
+let optionTwo = '';
+let votesOne = 0;
+let votesTwo = 0;
+let previousPolls = [];
+
+newPoll.addEventListener('submit', (e) => {
+    e.preventDefault();
+    const data = new FormData(newPoll);
+
+    const userQuestion = data.get('poll-question');
+    const userOptionOne = data.get('option-one-input');
+    const userOptionTwo = data.get('option-two-input');
+
+    question = userQuestion;
+    optionOne = userOptionOne;
+    optionTwo = userOptionTwo;
+
+    renderPoll();
+    displayCurrentPoll();
+});
+
+savePollBtn.addEventListener('click', () => {
+
+    const currentPollData = {
+        question: question,
+        optionOne: optionOne,
+        optionTwo: optionTwo,
+        votesOne: votesOne,
+        votesTwo: votesTwo
+    };
+
+    previousPolls.unshift(currentPollData);
```

Comment:
Why `.unshift()`? Adding something to the front of an array in some cases can be a lot of extra work for the computer because it needs to move everything in the array over to make room at the front. `.push()` adds items to the end of an array which is always going to be faster than adding something to the front. Just something to consider.